sycl/include/CL/sycl/marray.hpp

```diff
@@ -0,0 +1,355 @@
+//==----------------- marray.hpp --- SYCL types ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// Implements marray classes.
+
+#pragma once
+
+#include <CL/sycl/aliases.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/generic_type_traits.hpp>
+#include <CL/sycl/detail/type_traits.hpp>
+#include <CL/sycl/half_type.hpp>
+
+#include <array>
```

Comment:
It doesn't seem like array header is needed, could you please also clarify why other C++ standard headers are included?

Reply:
Not needed indeed